### PR TITLE
Guard against setting properties on tests that don't exist

### LINKED ISSUE #316: Local failure building tests -- test "web_client_HistomicsTK_annotations" does not exist
@jbeezley: @cdeepakroy and I get the error below when trying to build the tests on master. Clearly Travis has no problem with it, though. I get this error with a new build directory for Girder, and the particular revisions 6a7d593 and girder/girder@7b4022b. With `git bisect` I was able to locate bab64d8 as the first commit that causes the test build to fail.

```
-- Including plugin.cmake from "HistomicsTK"
CMake Error at plugins/HistomicsTK/plugin.cmake:100 (set_property):
  set_property given TEST names that do not exist:

    web_client_HistomicsTK_annotations

Call Stack (most recent call first):
  tests/CMakeLists.txt:111 (include)
  tests/CMakeLists.txt:123 (_add_plugin)


-- Configuring incomplete, errors occurred!
```

### PR DESCRIPTION
**Pull request by jbeezley**
Fixes #316

### CODE REVIEW

**Comment by cdeepakroy** [comment]
I confirm that this resolves the error!

File: plugin.cmake [hunk 1]
```diff
@@ -14,6 +14,11 @@
 #  limitations under the License.
 ###############################################################################
 
+# support the "if(TEST ...)" command
+if(POLICY CMP0064)
```
Comment (neal-siekierski) [inline plugin.cmake]: Will this work properly with older versions of CMake that don't have this policy?

Comment (jbeezley) [inline plugin.cmake]: For cmake < 3.4, the `if(TEST)` statement will be ignored.  This means the #316 will still occur, but using the defaults will still work.

Comment (neal-siekierski) [inline plugin.cmake]: All right. Would it be reasonable to make the error more transparent in the event that someone uses an old version of CMake and disables the tests, or are we not too concerned about that?

Comment (jbeezley) [inline plugin.cmake]: I'm not personally concerned about it, but I could address it.  One issue is that I can't think of a blanket way to improve the error message.  For this specific case, we could check a combination of cmake version + js testing option, but there are other possible reasons why a test would not be created by `add_web_client_test` or `add_python_test`.

Perhaps the best option is to add a new argument to our cmake functions to add custom environment variables.

Comment (neal-siekierski) [inline plugin.cmake]: If adding custom environment variables is something we'll want to keep doing, that seems like a good approach. Since these changes make for a quick fix for those of us on cmake>=3.4, if that's liable to be more complicated it might be worth doing as a separate PR.
